src/audio/copier/copier_dai.c

```diff
@@ -286,3 +286,29 @@ void copier_dai_free(struct copier_data *cd)
 		rfree(cd->dd[i]);
 	}
 }
+
+int copier_dai_prepare(struct comp_dev *dev, struct copier_data *cd)
+{
+	int ret;
+
+	if (cd->endpoint_num == 1) {
+		ret = dai_zephyr_config_prepare(cd->dd[0], dev);
+		if (ret < 0)
+			return ret;
+
+		ret = dai_zephyr_prepare(cd->dd[0], dev);
+		if (ret < 0)
+			return ret;
+	} else {
+		for (int i = 0; i < cd->endpoint_num; i++) {
+			ret = dai_zephyr_config_prepare(cd->dd[i], dev);
+			if (ret < 0)
+				return ret;
+
+			ret = dai_zephyr_prepare(cd->dd[i], dev);
+			if (ret < 0)
+				return ret;
+		}
```

Comment:
it seems to me like you don't need the `if`? Just use the loop and `if (cd->endpoint_num == 1)` the loop will just run once. Am I missing anything?